registration_test.go

```diff
@@ -247,3 +250,219 @@ func TestUnregister(t *testing.T) {
+	t.Helper()
+
+	if wsl.MockAvailable() {
+		panic("This function should be called with the real back-end only")
+	}
+
+	d = wsl.NewDistro(ctx, distroName)
+	_ = d.Unregister()
+
+	cmd := exec.CommandContext(ctx,
+		"wsl.exe",
+		"--install",
+		appxName,
+		"--no-launch",
+	)
+	out, err := cmd.Output()
+	require.NoErrorf(t, err, "could not install: %v. Stdout: %s", err, string(out))
+
+	defer wslExeGuard(time.Minute)()
+
+	//nolint:gosec // It's fine for tests
+	// Need to use powershell in order to find the launcher executable
+	cmd = exec.CommandContext(ctx,
+		"powershell.exe",
+		"-NonInteractive",
+		"-NoProfile",
+		"-NoLogo",
+		"-Command",
+		fmt.Sprintf("& %q install --root --ui=none", launcherName),
+	)
+	out, err = cmd.Output()
+	require.NoErrorf(t, err, "could not register: %v. Stdout: %s", err, string(out))
+
+	return d
+}
+
+//nolint:revive // No, I won't put the context before the *testing.T.
+func requireUninstallAppx(t *testing.T, ctx context.Context, appxName string) {
+	t.Helper()
+
+	if wsl.MockAvailable() {
+		panic("This function should be called with the real back-end only")
+	}
+
+	//nolint:gosec // It's fine for tests
+	cmd := exec.CommandContext(ctx,
+		"powershell.exe",
+		"-NonInteractive",
+		"-NoProfile",
+		"-NoLogo",
+		"-Command",
+		fmt.Sprintf("Get-AppxPackage %q | Remove-AppxPackage", appxName),
+	)
+	out, err := cmd.Output()
+	require.NoError(t, err, "could not clean up appx package. Stdout: %s", string(out))
+}
+
+//nolint:revive // No, I won't put the context before the *testing.T.
+func requireInstallFromAppxMock(t *testing.T, ctx context.Context, m *wslmock.Backend, appxName string, distroName string) (d wsl.Distro) {
+	t.Helper()
+
+	d = wsl.NewDistro(ctx, distroName)
+	err := d.Register(emptyRootFS)
+	require.NoError(t, err, "Setup: could not register")
+
+	guid, err := d.GUID()
+	require.NoError(t, err, "Setup: could not get GUID to access distro registry")
+
+	k, err := m.OpenLxssRegistry(fmt.Sprintf("{%s}", guid.String()))
+	require.NoError(t, err, "Setup: could not access distro registry")
+	defer k.Close()
+
+	//nolint:forcetypeassert // We know it's this type of key because we got it from the mock registry.
+	k.(*wslmock.RegistryKey).Data["PackageFamilyName"] = appxName
+	return d
+}
+
+// appxIsInstalled returns true if an AppxPackage is currently installed.
+func appxIsInstalled(ctx context.Context, appxName string) (bool, error) {
+	if wsl.MockAvailable() {
+		panic("This function is only available for real back-ends")
+	}
+
+	//nolint:gosec // Variable in test command is fine
+	cmd := exec.CommandContext(
+		ctx,
+		"powershell.exe",
+		"-NonInteractive",
+		"-NoProfile",
+		"-NoLogo",
+		"-Command",
+		fmt.Sprintf("(Get-AppxPackage -Name %q).Status", appxName),
+	)
+
+	out, err := cmd.Output()
+	if err != nil {
+		return false, fmt.Errorf("could not run Get-AppxPackage: %v. Stdout: %s", err, string(out))
+	}
+
+	if strings.Contains(string(out), "Ok") {
```

Comment:
ok, you have to keep that one :p

Reply:
Haha I almost messed it up with a find and replace 😅

Reply:
oh, you meant I missed otherwise having the chance to request for even more changes, damn :p